spec/controllers/applications_controller_spec.rb

```diff
@@ -0,0 +1,214 @@
+    end
+  end
+
+  describe 'get #NEW' do
+    subject { get :new }
+
+    context 'if user' do
+      before { sign_in_user }
+      it { expect(subject).to be_success }
+      it { is_expected.to have_http_status(200) }
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+      it { expect(subject).to be_success }
+      it { is_expected.to have_http_status(200) }
+    end
+
+    context 'if no user' do
+      it { expect(subject).not_to be_success }
+      it { is_expected.to have_http_status(302) }
+    end
+  end
+
+  describe 'get #SHOW' do
+    def show_app app
+      get :show, id: app.id
+    end
+
+    context 'if user' do
+      before { sign_in_user }
+
+      it 'should show user app' do
+        expect(show_app user_application).to have_http_status(200)
+      end
+
+      xit 'should not show admin app' do
+        expect(show_app admin_application).to have_http_status(302)
+      end
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+
+      it 'should show user app' do
+        expect(show_app user_application).to have_http_status(200)
+      end
+
+      it 'should show admin app' do
+        expect(show_app admin_application).to have_http_status(200)
+      end
+    end
+
+    context 'if no user' do
+      it 'should not show user app' do
+        expect(show_app user_application).to have_http_status(302)
+      end
+
+      it 'should not show admin app' do
+        expect(show_app admin_application).to have_http_status(302)
+      end
+    end
+  end
+
+  describe 'post #DESTROY' do
+    def destroy_app app
+      post :destroy, id: app.id
+    end
+
+    context 'if user' do
+      before { sign_in_user }
+
+      it 'should destroy user application' do
+        expect { destroy_app user_application }.to change { Application.count }.by -1
+      end
+
+      xit 'should not destroy admin application' do
+        expect { destroy_app admin_application }.to change { Application.count }.by 0
+      end
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+
+      it 'should destroy user application' do
+        expect { destroy_app user_application }.to change { Application.count }.by -1
+      end
+
+      it 'should destroy admin application' do
+        expect { destroy_app admin_application }.to change { Application.count }.by -1
+      end
+    end
+
+    context 'if no user' do
+      it 'should not destroy user application' do
+        expect { destroy_app user_application }.to change { Application.count }.by 0
+      end
+
+      it 'should not destroy admin application' do
+        expect { destroy_app admin_application }.to change { Application.count }.by 0
+      end
+    end
+  end
+
+  describe 'post #UPDATE' do
+    let(:app_attrs) { attributes_for(:application) }
+
+    def check_update(app, shloud_be_eq)
+      put :update, id: app.id, application: app_attrs
+      app = Application.find(app.id)
+      option = shloud_be_eq ? "to" : "not_to"
```

Comment:
Useless assignment to variable - `option`.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.